src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -114,6 +114,21 @@ options::find make_find_options(document::view arguments) {
     return options;
 }
 
+document::value bulk_write_result_to_document(const mongocxx::result::bulk_write& result) {
+    builder::basic::document upserted_ids_builder;
+    for (auto&& index_and_id : result.upserted_ids()) {
+        upserted_ids_builder.append(
+            kvp(std::to_string(index_and_id.first), index_and_id.second.get_int32().value));
+    }
+
+    return make_document(kvp("matchedCount", result.matched_count()),
```

Comment:
Per note [here](https://github.com/mongodb/mongo-cxx-driver/blob/57e37617421a8cb7210856ca491ce2131d9b4127/src/mongocxx/test/spec/operation.cpp#L1127), may want to include an `insertedIds: {}` field.